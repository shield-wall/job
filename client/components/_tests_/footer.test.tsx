import Footer from "../footer";
import { render } from '@testing-library/react';

describe("footer test", () => {
    render(<Footer />);
    const links = Array.from(document.querySelectorAll('#link'))
    const linksContent = links.map(a => a.innerHTML)

    it("is rendering footer's tag", () => {
        const footer = document.querySelector("#footer")
        expect(footer.getAttribute('id')).toBe('footer')
    })

    it.each(links)("is checking footer's links path exists", (a) => {

        expect(a.getAttribute('href')).toBeTruthy()

    })

    it("is rendering footer's links content", () => {
        links.forEach((a, i) => {
            expect(a.innerHTML).toBe(linksContent[i])
        })
    });
});