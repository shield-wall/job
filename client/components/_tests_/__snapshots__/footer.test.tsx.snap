// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer test is redendring properly footer component  1`] = `
<footer
  className="footer"
>
  <div
    className="container-full"
  >
    <span
      className="text-md text-gray-500"
    >
      © 2023
      <a
        className="hover:underline"
        href="/"
      >
        Job™
      </a>
      . All Rights Reserved.
    </span>
    <div
      className="flex mt-4 space-x-4 sm:justify-center sm:mt-0"
    >
      <a
        className="footer-item"
        href="/"
      >
        <i>
          Facebook
        </i>
      </a>
      <a
        className="footer-item"
        href="/"
      >
        <i>
          Instagram
        </i>
      </a>
      <a
        className="footer-item"
        href="/"
      >
        <i>
          Twitter
        </i>
      </a>
      <a
        className="footer-item"
        href="/"
      >
        <i>
          Github
        </i>
      </a>
    </div>
  </div>
</footer>
`;
